specification/vmware/resource-manager/Microsoft.AVS/stable/2023-03-01/skus.json

```diff
@@ -0,0 +1,381 @@
+            "create"
+          ]
+        },
+        "zones": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The availability zones of SKU location."
+        },
+        "zoneDetails": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SkuZoneDetail"
+          },
+          "description": "The availability zone details of the SKU location.",
+          "x-ms-identifiers": []
+        },
+        "extendedLocations": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The extended locations of SKU."
+        },
+        "type": {
+          "type": "string",
+          "enum": [
+            "Region",
+            "EdgeZone"
+          ],
+          "x-ms-enum": {
+            "name": "LocationType",
+            "modelAsString": true
+          },
+          "description": "Type of the extended location."
+        }
+      }
+    },
+    "Sku": {
+      "description": "The SKU definition.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the SKU."
+        },
+        "resourceType": {
+          "type": "string",
+          "description": "Resource type the SKU applicable for."
+        },
+        "tier": {
+          "type": "string",
+          "description": "This field is required to be implemented by the Resource Provider if the service has more than one tier, but is not required on a PUT."
+        },
+        "size": {
+          "type": "string",
+          "description": "The SKU size. When the name field is the combination of tier and some other value, this would be the standalone code. "
+        },
+        "family": {
+          "type": "string",
+          "description": "If the service has different generations of hardware, for the same SKU, then that can be captured here."
+        },
+        "kind": {
+          "type": "string",
+          "description": "If the service has different kinds of hardware, for the same SKU, then that can be captured here."
+        },
+        "locations": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "List of locations where this SKU is available."
+        },
+        "locationInfo": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SkuLocationAndZones"
+          },
+          "description": "List of locations where this SKU is available.",
+          "x-ms-identifiers": []
+        },
+        "capacity": {
+          "type": "object",
+          "items": {
+            "$ref": "#/definitions/SkuCapacity"
+          },
+          "description": "If the SKU supports scale out/in then the capacity integer should be included. If scale out/in is not possible for the resource this may be omitted."
```

Comment:
@cataggar This would be a breaking change as we are returning sku.capacity as an integer in master (refer [this](https://msazure.visualstudio.com/DefaultCollection/One/_git/Azure-Dedicated-AVS?path=/src/conveyor/conveyor/AzureFSharp/Domain.fs&version=GBmaster&line=496&lineEnd=496&lineStartColumn=7&lineEndColumn=29&lineStyle=plain&_a=contents)). In this case it conflicts with this change as it defines capacity as a dictionary rather than an integer. 

I believe we Should be keeping the ARM Sku contract of "Microsoft.AVS/skus" API separate from the Sku definition required in the PrivateCloud / Cluster PUT/PATCH requests?

It has been done in the similar fashion for other RPs like Microsoft.Compute, Microsoft.apimanagement, etc where they have separate Sku definitions for {providerNamespace}/skus and the other resource specific APIs.
cc: @amtyagi

Reply:
In the current `2022-05-01` API version, `Sku` is defined as:
https://github.com/Azure/azure-rest-api-specs/blob/237afa17929313c909f72f2fea59f4437e305c99/specification/vmware/resource-manager/Microsoft.AVS/stable/2022-05-01/vmware.json#L6521-L6533
There is no definition of `capacity`.
The `Sku` definition here is the [ARM Sku Contract](https://armwiki.azurewebsites.net/api_contracts/SKUContract.html?tabs=resource-type-level). It is the same definition as before, but with more optional fields.

Reply:
The link you provided is an internal implementation detail. Internally, it may make sense to use different types. We could restore `Sku` and rename the new type to `ResourceSku` [like Microsoft.Compute](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/compute/resource-manager/Microsoft.Compute/Skus/stable/2021-07-01/skus.json), but I'm not sure the benefit. Are you wanting to return `capacity` with an operation like `PrivateCloud_Get`? We don't currently return it.

Reply:
@cataggar, not with the PrivateCloud but we are going to return the capacity field as part of the Cluster_Get response with the Autoscale feature and also would be honouring the same in the Cluster PUT/PATCH calls. It would be better if we rename the new type to ResourceSku and extend the existing Sku type for the optional integer capacity field. Let me know if that makes sense and whether you are aligned with the thought. We can extend the existing Sku type in a separate swagger PR though.

Reply:
The  support for capacity field for Clusters is added in the September API version:

https://msazure.visualstudio.com/DefaultCollection/One/_git/Azure-Dedicated-AVS?path=/src/conveyor/conveyor/ConveyorHost/WebApi/Microsoft.AVS/ResourceDefinitions/PrivateClouds_Clusters/2023-09-01/Serialization.fs&version=GBmaster&line=180&lineEnd=180&lineStartColumn=22&lineEndColumn=70&lineStyle=plain&_a=contents

Reply:
I removed the sku changes d4a6b866c420b2805ff6f79307e55e19f8f7930c and we can flush out the details for the next API version.

Reply:
Sounds good. Will you be rebasing the September branch (https://github.com/Azure/azure-rest-api-specs-pr/tree/AVS/2023-09-01) with these changes so that we can introduce the Sku changes required for Autoscale?